### DISCUSSION TITLE
Handling Unexpected Tab Closure and Navigation Issues with SeleniumBase

### DISCUSSION THREAD

**Discussion by benfrankie** [seleniumbase/SeleniumBase#2330] [Q&A] [answered]
Hi, 

I am using SeleniumBase for web automation and have encountered an issue where occasionally driver.get(url) results in the current tab being closed and a new tab being opened. This behavior causes a loss of the previous page's context, including its scroll position. I attempted to use driver.execute_script("window.history.go(-1)") to navigate back, but it doesn't work if the original tab is closed. Is this a known behavior with SeleniumBase, or could it be related to specific browser settings or website behavior? 

Also, is there a way to ensure navigation happens within the same tab and to retain the scroll position of the previous page?

Thanks!

**Comment by mdmintz** [comment, marked as answer]
Hello. What you're describing is the default behavior in UC Mode when you use `driver.get(url)` to open a URL that has bot-detection software running on it. The current tab is replaced with a clean one (that doesn't have any traces of Selenium from within the Chrome browser console) and the new URL is opened.

You can use `driver.default_get(url)` to open the new URL in the current tab without using UC Mode. Or you can try `driver.uc_open(url)`, which has better anti-detection mechanisms than `driver.default_get(url)`, but it's not as good as `driver.get(url)` / `driver.uc_open_with_tab(url)` / `driver.uc_open_with_reconnect(url, reconnect_time)` (which close the current tab and open a new one).

This is all covered in:
<b>https://www.youtube.com/watch?v=5dMFI3e85ig</b>

<a href="https://www.youtube.com/watch?v=5dMFI3e85ig"><img src="https://github.com/seleniumbase/SeleniumBase/assets/6788579/4c9a112a-25c6-458f-83c6-c4b910b8b636" title="undetected automation" width="330" /></a>

**Reply by benfrankie** [reply]
Is there a recommended SeleniumBase method or practice for handling scenarios where navigating back to a previous page requires re-locating elements to avoid stale element references, especially in a script that involves frequent navigation between pages? @mdmintz 

I apologize if this was also covered in your video – I watched it a couple times but might have been a bit distracted by the Star Wars clips. I'm still new to the technical aspects and currently learning with the help of ChatGPT. :-) Thank you!

**Reply by mdmintz** [reply]
Every time you navigate to a page, (even if going back to a page you were on before), you have to find / re-find the element. Any elements you found from a previous page load will lead to a `StaleElementReferenceException` if you try to use those elements without re-finding them first.
